Assets/Mirror/Core/NetworkManager.cs

```diff
@@ -215,27 +247,27 @@ public virtual void Awake()
         // virtual so that inheriting classes' Start() can call base.Start() too
         public virtual void Start()
         {
-            // headless mode? then start the server
-            // can't do this in Awake because Awake is for initialization.
-            // some transports might not be ready until Start.
+            // Auto-start headless server or client.
             //
-            // (tick rate is applied in StartServer!)
+            // We can't do this in Awake because Awake is for initialization
+            // and some transports might not be ready until Start.
             //
-            // don't auto start in editor where we have a UI, only in builds.
-            // otherwise if we switch to 'Dedicated Server' target and press
-            // Play, it would auto start the server every time.
-            if (Utils.IsHeadless() && !Application.isEditor)
+            // Note 1: It is intentional that selecting Dedicated Server in the
+            //         editor and clicking Play starts a server or client, depending
+            //         on the headlessStartMode. This is useful for debugging.
+            //
+            // Note 2: sendRate is applied in StartServer
+#if UNITY_SERVER
```

Comment:
please keep the runtime .isheadless check, that would be a discussion for a separate pr

Reply:
done